### PR TITLE
Update useBlockExplorer to be multi chain compatible. Remove need for envs `NEXT_PUBLIC_BLOCKCHAIN_EXPLORER_URL` and `NEXT_PUBLIC_BLOCKCHAIN_EXPLORER_NAME`

### PR DESCRIPTION
**Pull request by antho1404**
### Description

Update the `useBlockExplorer` hook to accept a `chainId` and return the proper network based on this chain ID.

I had to create 2 new buttons `AcceptOffer` and `CancelOffer` as the modal was outside of an iteration, and I couldn't get the chain id to display the proper network.

### How to test

Create an offer, a bid, accept an offer, create an nft... and make sure that all the modals have the link to etherscan

### CODE REVIEW

**Review by NicolasMahe — APPROVED** [review]

File: components/Button/AcceptOffer.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }, [accept, onClose, onOpen, onAccepted, offer, quantity, onError])
+
+  return (
+    <>
+      <ButtonWithNetworkSwitch
+        chainId={chainId}
+        {...props}
+        disabled={activeStep !== AcceptOfferStep.INITIAL}
+        isLoading={activeStep !== AcceptOfferStep.INITIAL}
```
Comment (ismailToyran) [inline components/Button/AcceptOffer.tsx:56]: We shouldn't use `disabled`, instead use `isDisabled`
But in this case, as we already have `isLoading` we shouldn't even use `isDisabled`

File: components/Button/CancelOffer.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }, [cancel, onClose, onOpen, onCanceled, offerId, onError])
+
+  return (
+    <>
+      <ButtonWithNetworkSwitch
+        chainId={chainId}
+        {...props}
+        disabled={activeStep !== CancelOfferStep.INITIAL}
+        isLoading={activeStep !== CancelOfferStep.INITIAL}
```
Comment (ismailToyran) [inline components/Button/CancelOffer.tsx:50]: We shouldn't use `disabled`, instead use `isDisabled`
But in this case, as we already have `isLoading` we shouldn't even use `isDisabled`

File: components/Button/AcceptOffer.tsx (hunk 1 above)
Comment (NicolasMahe) [reply components/Button/AcceptOffer.tsx:56]: @ismailToyran can you make the fix in a new PR please?

File: components/Button/CancelOffer.tsx (hunk 2 above)
Comment (NicolasMahe) [reply components/Button/CancelOffer.tsx:50]: @ismailToyran can you make the fix in a new PR please?

File: components/Button/AcceptOffer.tsx (hunk 1 above)
Comment (ismailToyran) [reply components/Button/AcceptOffer.tsx:56]: https://github.com/liteflow-labs/starter-kit/pull/201

File: components/Button/CancelOffer.tsx (hunk 2 above)
Comment (ismailToyran) [reply components/Button/CancelOffer.tsx:50]: https://github.com/liteflow-labs/starter-kit/pull/201

**Pull request merged**